service/dynamodb/dynamodbattribute/decode.go

```diff
@@ -449,7 +449,20 @@ func (d *Decoder) decodeMap(avMap map[string]*dynamodb.AttributeValue, v reflect
 		fields := unionStructFields(v.Type(), d.MarshalOptions)
 		for k, av := range avMap {
 			if f, ok := fieldByName(fields, k); ok {
-				fv := v.FieldByIndex(f.Index)
+				fv := v
+				for i, x := range f.Index {
```

Comment:
@jasdel, I'm thinking of something as follows:
```
// It breaks if OnEmbeddedNilStruct returns false
FieldByIndex(index, func OnEmbeddedNilStruct(v reflect.Value) bool)

// Here in decode.go:
fv := FieldByIndex(v, func (v reflet.Value) bool {
    v.Set(reflect.New(f.Type().Elem()))
    return true // to continue the loop.
}

// And in encode.go:
fv := FieldByIndex(v, func (v reflet.Value) bool {
    found = false
    return false // to break the loop.
}
```
Not sure if this is what you want (it sounds a bit obscure to me indeed).

And if you want it, where would you suggest me putting the function to.